tests/infrastructure/golden_images/update_boot_source/utils.py

```
@@ -313,3 +314,48 @@ def wait_for_data_source_unchanged_referenced_volume(data_source: DataSource, vo
+        check_exact_match: If True, wait for volume reference to match volume_name exactly.
+            If False, wait for volume reference to change from volume_name (the previous value).
+        volume_name: When check_exact_match=True, the expected volume name.
+            When check_exact_match=False, the previous volume name that should change.
+
+    Raises:
+        TimeoutExpiredError: If the expected condition is not met within 10 minutes.
+    """
+
+    if check_exact_match:
+
+        def condition():
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `condition`

(ANN202)